### ISSUE TITLE
automatically call a uniswap smart contract

### ISSUE THREAD

**Issue by electroxts** [graphprotocol/mission-control-curator#120]
is it possible to write a Smart contract to automatically call a uniswap eth / dai to pay 10 dai when adding a new project to Everest? swap call, write-off and addition, for convenience and gas saving, pack in one transaction
For example 
https://www.tokensets.com/portfolio/dpi
https://docs.tokensets.com/protocol/untitled
implementation of such

**Comment by Maksim3110** [comment]
super, very necessary. And you don't need to make exchanges on third-party services

**Comment by kovrus3080** [comment]
it would be very convenient and save a lot of time

**Comment by Cryptomilion** [comment]
good idea, no need to waste time exchanging

**Comment by Vladimir0122** [comment]
good idea

**Comment by GostFarest** [comment]
It's a good idea!
Out of inexperience, I thought that the 10DAI swap would happen automatically. Only after the failure I realized that DAI needs to be bought by yourself)

**Comment by itsjerryokolo** [comment]
This would make adding projects efficient. Love it.

**Comment by nefed007** [comment]
I like your train of thought. continue)

**Comment by koen84** [comment]
> And you don't need to make exchanges on third-party services

uniswap IS a third-party service.

**Comment by electroxts** [comment]
> > And you don't need to make exchanges on third-party services
> 
> 
> 
> uniswap IS a third-party service.

 I have indicated uniswap as an example, you can use other DEXs

**Comment by electroxts** [comment]
Thank you very much for your feedback and suggestions, I am very glad that you liked my idea, I hope the team will also like it and implement it, I will be proud!

**Comment by yanivtal** [comment]
Seems like a lot of people like this idea. Let's coordinate to get it done.

**Comment by electroxts** [comment]
> Seems like a lot of people like this idea. Let's coordinate to get it done. 

good idea, it remains to find someone who implements it into production 👍

**Comment by AlexKlos** [comment]
> > Seems like a lot of people like this idea. Let's coordinate to get it done.
> 
> good idea, it remains to find someone who implements it into production 👍

I think this might work. This is a [simple smart contract](https://github.com/AlexKlos/ApplySignedWithAttributeAndPermitExtended), but it calls a whole bunch of other contracts when it works, so I can't test it properly.

**Comment by electroxts** [comment]
@yanivtal help with testing 👆

**Comment by kovrus3080** [comment]
the guys from The Graph team could help test the smart contract, or we will have to look for other specialists
